flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -1035,6 +1030,14 @@ protected void registerCachedFilesWithPlan(Plan p) throws IOException {
 		}
 	}
 
+	/**
```

Comment:
This shouldn't be exposed. You can write the test strictly without using an `ExecutionEnvironment`. Which in the end users of `PlanGenerator` probably also have to do.

Reply:
consider the following user table program:
val env = ExecutionEnvironment.getExecutionEnvironment()
val tableEnv = BatchTableEnvironment.create(env, config)
env.registerCachedFile("/tmp/cache", "my_cache")
val input = env.fromElements(...)...
tEnv.fromDataSet(input)...
tEnv.execute() // will generate plan through PlanGenerator in this method

If we do not expose this method, the generated plan does not contain the cache files registered in DataSet program.(which can be converted to Table through `BatchTableEnvironment.fromDataSet` method)

Reply:
True, but I think this is tricky, because there could be any number of options that we want to forward from the `ExecutionEnvironment` but that are easy to forget. If there is a general way of forwarding everything that would probably be ok.

Reply:
I find `ExecutionConfig` has forwarded everything except `cacheFile`. CMIIW

Reply:
hi @aljoscha, do you have any other concern about this ?

Reply:
From the discussion, it seems that this method was added:
1) for the tests 
2) for future usage in the `TableEnvironment`.

If this is correct, I would suggest to remove it from this PR and add it to the PR that uses it. Now it seems that this is an unused `public` method.

In addition, I agree with @aljoscha 's concerns that we may need to come up with a better and more foolproof design on how to forward stuff from the `ExecutionEnvironment` to the `TableEnvironment`. So if we achieve that, then this method would be instant legacy. 

WDYT?

Reply:
Make sense to me, I will revert it